holoviews/core/data/spatialpandas.py

```diff
@@ -616,31 +616,35 @@ def get_value_array(data, dimension, expanded, keep_index, geom_col,
     Returns:
         An array containing the values along a dimension
     """
+    if not len(data):
+        return np.array([])
     column = data[dimension.name]
     if keep_index:
         return column
-    all_scalar = True
+    is_scalars = [isscalar(value) for value in column]
+    all_scalar = all(is_scalars)
+    if all_scalar and not expanded:
+        scal_unique = column.unique()
+        # .unique() on categorical dtype doesn't return a np array
+        return scal_unique if isinstance(scal_unique, np.ndarray) else scal_unique.to_numpy()
```

Comment:
I would assume `column.unique()` would return a pd.Series (or maybe spd.GeoSeries), where the underlying values can be a numpy array. 

Can you check that the instance check is not always false?

Reply:
`pd.Series.unique()` returns an ndarray or an ExtensionArray (https://pandas.pydata.org/docs/reference/api/pandas.Series.unique.html) and that's what I observed with a categorical column (https://github.com/holoviz/holoviews/pull/6470#issuecomment-2526317654). Does that answer your question?

Reply:
It does